src/wp-includes/media.php

```diff
@@ -5449,3 +5449,283 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since 1.2.0
+ *
+ * @param array $metadata      The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ * @return array $metadata The attachment metadata.
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	$dominant_color_data = _dominant_color_get_dominant_color_data( $attachment_id );
+	if ( ! is_wp_error( $dominant_color_data ) ) {
+		if ( isset( $dominant_color_data['dominant_color'] ) ) {
+			$metadata['dominant_color'] = $dominant_color_data['dominant_color'];
+		}
+
+		if ( isset( $dominant_color_data['has_transparency'] ) ) {
+			$metadata['has_transparency'] = $dominant_color_data['has_transparency'];
+		}
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since 1.2.0
+ *
+ * @param array  $attr       Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ * @return mixed $attr Attributes for the image markup.
+ */
+function dominant_color_update_attachment_image_attributes( $attr, $attachment ) {
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	if ( isset( $image_meta['has_transparency'] ) ) {
+		$attr['data-has-transparency'] = $image_meta['has_transparency'] ? 'true' : 'false';
+
+		$class = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
+		if ( empty( $attr['class'] ) ) {
+			$attr['class'] = $class;
+		} else {
+			$attr['class'] .= ' ' . $class;
+		}
+	}
+
+	if ( ! empty( $image_meta['dominant_color'] ) ) {
+		$attr['data-dominant-color'] = esc_attr( $image_meta['dominant_color'] );
+		if ( empty( $attr['style'] ) ) {
+			$attr['style'] = '';
+		}
+		$attr['style'] .= '--dominant-color: #' . esc_attr( $image_meta['dominant_color'] ) . ';';
+	}
+
+	return $attr;
+}
+add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_update_attachment_image_attributes', 10, 2 );
+
+/**
+ * Filter image tags in content to add the dominant color to the image.
+ *
+ * @since 1.2.0
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context        The context of the image.
+ * @param int    $attachment_id  The attachment ID.
+ * @return string image tag
+ */
+function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $attachment_id ) {
```

Comment:
Let's not filter and manaully add this in here - https://github.com/WordPress/wordpress-develop/blob/f52d476bc6dcfe4c5d011f335361500f80708620/src/wp-includes/media.php#L1854